test/rules/std_test.py

```diff
@@ -25,7 +25,6 @@
         # NB The newline before the comma, should report on the comma, not the newline
         # for clarity.
         ("L005", "whitespace_errors.sql", [(2, 9)]),
-        ("L019", "whitespace_errors.sql", [(4, 1)]),
```

Comment:
What happened here?

Reply:
Ok - this is a good point to discuss.

That file effectively has a part which has this structure if you simplify it:
```sql
SELECT
    a
    , -- comment
    b
```

Previously that comma line would get flagged up as a _leading_ comma and so trigger `L019`.

In a situation like this there are 4 possible cases:
1. `inline` (i.e. no newlines either side): `a, b`
2. `leading`: `a\n, b` (newline before comma, not after)
3. `trailing`: `a,\n b` (newline after comma, not before)
4. `alone`: `a\n,\n b` (newlines both before _and_ after)

That line in `whitespace_errors.sql` is a situation which I've classed as `alone` (even though there's a comment on that line too which I think adds to the ambiguity).

I got a bit deeper into this in the next PR, which I think I'll bring more of into this PR, but where I'm aiming at is that if you configure yourself for `trailing` (the default), then it also allows `inline` _and_ `alone`, for commas this might initially seem weird but for operators I think it's easier to picture and I don't see why commas should be different. The extension of this is that I'm thinking to allow users to qualify that configuration with `:strict` which means the other cases aren't allowed anymore.

Given that I'm very slightly changing the behaviour of `L019` in doing this - this particular test no longer fails for L019 because I'm not allowing this case by default.

Do you think this is a sensible move?

Reply:
I think so. It's natural that as we think more deeply and rigorously about some of these corner cases, we'll refine it beyond what we did in the first version. 👍